server/src/routes/current-user.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * A router that serves the /api/auth/currentuser endpoint
+ * This route returns information about the current user that is logged in
+ * It uses the currentUser middleware which creates a currentUser property on request or 'req' object
+ * It also uses the requireAuth middleware which checks if the user is logged in or not
+ */
+
+const router = require("express").Router();
+require("dotenv").config();
+const currentUser = require("../middleware/current-user");
+const requireAuth = require("../middleware/require-auth");
+
+router.get(
+  "/api/auth/currentuser",
+  currentUser, //A middleware to check for currentUser stored inside of jwt value of cookie and assign it to a property of currentUser on the request object
+  requireAuth, //A middleware to confirm that a user is logged in otherwise would throw a Not authorized error
```

Comment:
I feel like we will always use currentUser and requireAuth together right? I can't see a use case when we don't. Should we combine those middlewares to make things a bit more cleaner?